services/src/main/java/org/keycloak/services/resources/admin/AuthenticationManagementResource.java

```diff
@@ -599,7 +648,70 @@ public void updateExecutions(@PathParam("flowAlias") String flowAlias, Authentic
             model.setRequirement(AuthenticationExecutionModel.Requirement.valueOf(rep.getRequirement()));
             realm.updateAuthenticatorExecution(model);
             adminEvent.operation(OperationType.UPDATE).resource(ResourceType.AUTH_EXECUTION).resourcePath(session.getContext().getUri()).representation(rep).success();
+            //return;
+            return Response.accepted(flow).build();
+        }
+
+        //executions can't have name and description updated
+        if (rep.getAuthenticationFlow() == null) { return Response.accepted(flow).build();}
+
+        //check if updating a correct flow
+        AuthenticationFlowModel checkFlow = realm.getAuthenticationFlowById(rep.getFlowId());
+        if (checkFlow == null) {
+            session.getTransactionManager().setRollbackOnly();
+            throw new NotFoundException("Illegal execution");
+        }
+
+        //if a different flow with the same name does already exist, throw an exception
+        if (realm.getFlowByAlias(rep.getDisplayName()) != null && !checkFlow.getAlias().equals(rep.getDisplayName())) {
+            return ErrorResponse.exists("Flow alias name already exists");
+            //throw new BadRequestException("Name already exists");
+        }
+
+        //if the name changed
+        if (!checkFlow.getAlias().equals(rep.getDisplayName())) {
+            checkFlow.setAlias(rep.getDisplayName());
+        }
+
+        //check if the description changed
+        if (!checkFlow.getDescription().equals(rep.getDescription())) {
+            checkFlow.setDescription(rep.getDescription());
+        }
+
+        //update the flow
+        realm.updateAuthenticationFlow(checkFlow);
+        adminEvent.operation(OperationType.UPDATE).resource(ResourceType.AUTH_EXECUTION).resourcePath(session.getContext().getUri()).representation(rep).success();
+        return Response.accepted(flow).build();
+
+/*        AuthenticationFlowModel checkFlow = realm.getFlowByAlias(rep.getDisplayName());
```

Comment:
Maybe remove this comment?

Reply:
I left it there deliberately. It is now removed